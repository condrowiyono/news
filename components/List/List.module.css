.container {
  text-decoration: none;
  width: 100%;
}

.header {
  display: flex;
  margin-bottom: 6px;
  justify-content: space-between;
}

.date {
  font-weight: 400;
  font-size: 11px;
  line-height: 140%;
  color: #797D7F;
  margin-bottom: 6px;
}

.title {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;  
  overflow: hidden;
  font-weight: 900;
  font-size: 16px;
  line-height: 130%;
  color: #14171A;
}

.thumbnail {
  flex-shrink: 0;
  margin-left: 16px;
  position: relative;
  width: 112px;
  height: 84px;
}

.watermark {
  padding: 4px 13px;
  border-top-left-radius: 8px;
  border-bottom-right-radius: 8px;
  position: absolute;
  right: 0;
  bottom: 0;
  background: #FFFFFF;
}

.footer {
  font-weight: 400;
  font-size: 12px;
  line-height: 160%;
  color: #797D7F;
  display: flex;
  justify-content: space-between;
}

.category {
  text-transform: capitalize;
}