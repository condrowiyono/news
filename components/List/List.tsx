import style from "./List.module.css";
import ImageServer from "@components/Image";
import Link from "next/link";
import { formatDate } from "@utils/intl";
import Skeleton from "@components/Skeleton";

export const SkeletonList = () => {
  return (
    <div className={style.container}>
      <div className={style.header}>
        <div data-testid="title">
          <Skeleton width={72} height={12} />
          <Skeleton width={200} height={24} />
        </div>
        <Skeleton width={112} height={84} />
      </div>
      <div className={style.footer}>
        <Skeleton width={72} height={12} />
        <button className={style.share}>
          <ImageServer
            src="/icons/ic_share.png"
            width={20}
            height={20}
            alt="share-button"
          />
        </button>
      </div>
    </div>
  );
};

const List = ({ image, title, date, category, link = "#" }: INews) => {
  return (
    <Link href={link}>
      <a className={style.container}>
        <div className={style.header}>
          <div>
            <div className={style.date}>
              {formatDate.format(new Date(date))}
            </div>
            <div className={style.title} data-testid="title">
              {title}
            </div>
          </div>
          <div className={style.thumbnail}>
            <ImageServer
              src={image}
              objectFit="cover"
              width={112}
              height={84}
              alt={title}
              style={{ borderRadius: 8 }}
            />
          </div>
        </div>
        <div className={style.footer}>
          <div className={style.category}>{category}</div>
          <button className={style.share}>
            <ImageServer
              src="/icons/ic_share.png"
              width={20}
              height={20}
              alt="share-button"
            />
          </button>
        </div>
      </a>
    </Link>
  );
};

export default List;
