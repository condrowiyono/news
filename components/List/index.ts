import List, { SkeletonList } from "./List";

export { SkeletonList };
export default List;