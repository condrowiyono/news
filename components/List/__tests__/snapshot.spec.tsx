import { render } from "@testing-library/react";
import List, { SkeletonList } from "@components/List";
import { news } from "./List.spec";

describe("Snapshot List", () => {
  it("renders List unchanged", () => {
    const { container } = render(<List {...news} />);
    expect(container).toMatchSnapshot();
  });

  it("renders SkeletonList unchanged", () => {
    const { container } = render(<SkeletonList />);
    expect(container).toMatchSnapshot();
  });
});
