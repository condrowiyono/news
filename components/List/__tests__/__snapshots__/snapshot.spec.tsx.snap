// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot List renders List unchanged 1`] = `
<div>
  <a
    class="container"
    href="/#"
  >
    <div
      class="header"
    >
      <div>
        <div
          class="date"
        >
          2 Agustus 2022 08.12
        </div>
        <div
          class="title"
          data-testid="title"
        >
          Judul
        </div>
      </div>
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27112%27%20height=%2784%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Judul"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="border-radius: 8px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
          />
          <noscript />
        </span>
        <div
          class="watermark"
        >
          <img
            alt="logo-idn"
            height="16"
            src="/logos/logo_idntimes.png"
            width="45"
          />
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="category"
      >
        Bisnis
      </div>
      <button
        class="share"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="share-button"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
          />
          <noscript />
        </span>
      </button>
    </div>
  </a>
</div>
`;

exports[`Snapshot List renders SkeletonList unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <div
        data-testid="title"
      >
        <div
          class="skeleton"
          data-testid="skeleton"
          style="height: 12px; width: 72px; border-radius: 4px;"
        />
        <div
          class="skeleton"
          data-testid="skeleton"
          style="height: 24px; width: 200px; border-radius: 4px;"
        />
      </div>
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 84px; width: 112px; border-radius: 4px;"
      />
    </div>
    <div
      class="footer"
    >
      <div
        class="skeleton"
        data-testid="skeleton"
        style="height: 12px; width: 72px; border-radius: 4px;"
      />
      <button
        class="share"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2720%27%20height=%2720%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="share-button"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(data:image/gif;base64,R0lGODlhAQABAPAAAL29vf///yH5BAAAAAAALAAAAAABAAEAAAICRAEAOw==);"
          />
          <noscript />
        </span>
      </button>
    </div>
  </div>
</div>
`;
